AI ChatBot Added

Nishitbaria opened the pull request:
## Related Issue


Closes: #210


## Description of Changes

This pull request introduces an AI Voice and Chat Assistant to the repository, enhancing the user experience by providing support for Frequently Asked Questions (FAQ), page navigation, and overall user interaction.

## Changes Made

1. Integrated an AI voice and chat assistant feature using a suitable library or API to provide interactive support to users.
2. Added necessary JavaScript code to initialize and handle the AI voice and chat assistant on relevant pages.
3. Created a new FAQ section, either as a separate page or as an overlay, to address common user queries and provide instant responses using the AI assistant.
4. Designed an intuitive and user-friendly interface for the chat assistant, ensuring smooth conversation flow and easy access to information.
5. Configured the chat assistant to handle user queries related to page navigation, helping users quickly find the desired information or sections within the website.
6. Implemented appropriate error handling and fallback mechanisms in case the AI assistant encounters any issues or fails to provide satisfactory responses.
7. Conducted thorough testing to ensure the voice and chat assistant functions correctly, responds accurately to user queries, and enhances the overall user experience.
8. Updated relevant documentation or README files to provide instructions on how to interact with the AI voice and chat assistant and make the most of its capabilities.



## Testing Instructions

1. Clone this branch and navigate to the project directory.
2. Open the relevant page(s) where the AI voice and chat assistant is implemented.
3. Ensure that the chat assistant interface is visible and easily accessible on the page.
4. Test various user queries related to the FAQ section or page navigation and verify that the AI assistant provides accurate and helpful responses.
5. Verify that the chat assistant handles unexpected queries gracefully and provides appropriate error messages or fallback responses.
6. Test the voice functionality, if implemented, and ensure that the AI assistant responds to voice commands or questions effectively.
## Checklist:

<!--
Mark the checkboxes to indicate completion. Example: 
- [x] My code follows the style guidelines of this project.
-->

- [x] My code adheres to the established style guidelines of this project.
- [x] I have conducted a self-review of my code.
- [x] I have included comments in areas that may be difficult to understand.
- [ ] I have made corresponding updates to the project documentation.
- [x] My changes have not introduced any new warnings.


## Screenshots

|      Original       |       Updated        |
| :-----------------: | :------------------: |
| ![Original screenshot](https://github.com/apoorvaron/Shorty/assets/85815172/da3d63ac-424f-498f-b472-0b3d6fa23626)| ![Updated screenshot](https://github.com/apoorvaron/Shorty/assets/85815172/4c9d6f19-8e55-40cf-95ac-f3d43279dd2b) |

Checkout the video:

Please provide any necessary screenshots to illustrate the changes made.

Nishitbaria commented:
@apoorvaron The routes for navigation are set according to local and will update as soon as you will merge

apoorvaron commented:
@Nishitbaria 

1) Alignment issue (overlap with chatbot and scroll to top ) 
2) Theme matching?  (light blue color or Aibot and dark blue of shorty)
3) login redirection link is wrong.

Nishitbaria commented:
@apoorvaron ,can you  explain first point with image

apoorvaron commented:
@Nishitbaria 

<img width="354" alt="Screenshot 2023-07-05 at 7 08 43 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/a1f560f2-18ac-4711-98fb-ab584255b993">

apoorvaron commented:
@Nishitbaria 

1) Sometimes after refresh, opens without clicking bot
2) Rename to Shorty Chatbot 

<img width="160" alt="Screenshot 2023-07-11 at 4 20 17 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/f743ee20-5eb0-4531-bc1d-8089561f0fe9">

3) What if in future , i want to add personalized questions?

4) Dimensions of box are not fixed, seems different after refreshing and clicking

<img width="1440" alt="Screenshot 2023-07-11 at 4 21 18 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/5772c77f-4a9b-4ee0-9854-d264bb9ff174">

5) Use dark shades of blue , and share screenshots (may be u have to revert to light blue)

6) don't remove current bot , will remove if this works perfectly fine on hosting 

7) Alignment issue 

<img width="353" alt="Screenshot 2023-07-11 at 4 23 49 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/ec02c9a4-3246-47a8-86aa-347fc8f6826c">

Nishitbaria commented:
@apoorvaron 
1. I Will rename the Chatbot
2. The Chatbot textbox size varies means user can change is by dragging
3. Give me color HexCode you want to implement
4. Regarding to personlize question as you merge this I will give you required Script and Guidance so you can host this and do changes regarding it

Nishitbaria commented:
@apoorvaron What Alignment Issues  ?

apoorvaron commented:
> @apoorvaron
> 
> 1. I Will rename the Chatbot
> 2. The Chatbot textbox size varies means user can change is by dragging
> 3. Give me color HexCode you want to implement
> 4. Regarding to personlize question as you merge this I will give you required Script and Guidance so you can host this and do changes regarding it

1) Ok 
2) But Sometimes i refresh the page and opening the chatbot, its dimensions are different (without dragging )
3) Try at your end , will review after 
4) Ok 
5)  

https://github.com/apoorvaron/Shorty/assets/83576954/ef5a451a-4a6a-4607-bee6-246f5f037fd2

apoorvaron commented:
@Nishitbaria Done changes?

Nishitbaria commented:
@apoorvaron 

I have changed Name and Background Color also and also Fixed Size Issue

![image](https://github.com/apoorvaron/Shorty/assets/85815172/99514842-1e93-43c3-ae65-d601588a2db1)

Nishitbaria commented:
Please Review it and merge we will solve further issues in another PR

apoorvaron commented:
@Nishitbaria  can you change the icon ( replace with current DB chatbot icon )

<img width="96" alt="Screenshot 2023-07-16 at 12 33 31 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/8824c3cb-9607-4a56-8469-f3b71c0d84fd">

Nishitbaria commented:
![image](https://github.com/apoorvaron/Shorty/assets/85815172/0a5a0317-2d77-4bc6-9e43-895a176d337d)

Its Paid Feature we can customize  color of this icon but cant replace in this basic/developer plan

Nishitbaria commented:
@apoorvaron Removed old Chatbot

apoorvaron commented:
@Nishitbaria 

1) I din't tell you to remove scroll to top btn ( and do same styling as AIbot ex: round border & radius  )
2) AIbot must align to right 


<img width="1439" alt="Screenshot 2023-07-16 at 11 25 51 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/da067e14-ae27-4399-9dc5-b5403b460c17">

Nishitbaria commented:
> @Nishitbaria
> 
> 1. I din't tell you to remove scroll to top btn ( and do same styling as AIbot ex: round border & radius  )
> 2. AIbot must align to right
> 
> <img alt="Screenshot 2023-07-16 at 11 25 51 PM" width="1439" src="https://user-images.githubusercontent.com/83576954/253819427-da067e14-ae27-4399-9dc5-b5403b460c17.png">

Damn !!!! 🤣🤣🤣 I will fix

Nishitbaria commented:
@apoorvaron ,I have made changes according to your suggestion ,please review and  merge my pull request after all its position can be changes by user by dragging and its fully fucntionalional chatbot with FAQ,navigation and manh other features willl be added in Future

Nishitbaria commented:
@apoorvaron Thanks Man

Nishitbaria commented:
I will make pr of other issues in 2 days So get ready to review it

apoorvaron commented:
Okk
